pkg/diagnostics/grpc_tracing.go

```diff
@@ -29,6 +30,12 @@ func SetTracingInGRPCMiddlewareUnary(appID string, spec config.TracingSpec) grpc
 		var resp interface{}
 		var span *trace.Span
 
+		// do not start the client/server spans, just set the trace context when the sampling rate is 0
+		if !diag_utils.IsTracingEnabled(spec.SamplingRate) {
+			resp, err = handler(newCtx, req)
```

Comment:
this is wrong. in this case, all requests will be discarded. handler must be called all the time. 

the other code needs to be flagged.

Reply:
handler is called in both the cases. please check.

Reply:
Can we call handler in one place ?

Reply:
I chose/preferred that way so bail out early based on 'span or not span' in code blocks . also context is different in calling handler.